src/main/java/com/commercetools/sync/products/helpers/ProductReferenceResolver.java

```diff
@@ -432,6 +432,6 @@ public ProductReferenceResolver(@Nonnull final ProductSyncOptions productSyncOpt
         }
 
         return collectionOfFuturesToFutureOfCollection(futures, toList())
-            .thenApply(maps -> productKeys.isEmpty() ? Collections.emptyMap() : maps.get(0));
+            .thenCompose(ignored -> productService.cacheKeysToIds(Collections.emptySet())); // return all cache.
```

Comment:
I guess it's the time to introduce a getter for the caches in the service layer. wdyt?

Reply:
For now, the cache is an internal part of the service, for this PR, I will focus the part of the quickly fixing the issues with returning cache. We will improve the caching with different PR soon.

Reply:
Added as a context: https://github.com/commercetools/commercetools-sync-java/issues/582